xlators/cluster/dht/src/dht-common.c

```diff
@@ -11195,22 +11196,28 @@ dht_notify(xlator_t *this, int event, void *data, ...)
 }
 
 int
-dht_inode_ctx_layout_get(inode_t *inode, xlator_t *this, dht_layout_t **layout)
+dht_inode_ctx_layout_get(inode_t *inode, xlator_t *this, dht_layout_t **layout,
+                         int ref)
 {
     dht_inode_ctx_t *ctx = NULL;
     int ret = -1;
+    uint64_t ctx_int = 0;
 
-    ret = dht_inode_ctx_get(inode, this, &ctx);
-
-    if (!ret && ctx) {
-        if (ctx->layout) {
-            if (layout)
-                *layout = ctx->layout;
-            ret = 0;
-        } else {
-            ret = -1;
+    LOCK(&inode->lock);
+    {
+        ret = __inode_ctx_get(inode, this, &ctx_int);
+        if (!ret) {
+            ctx = (dht_inode_ctx_t *)(uintptr_t)ctx_int;
+            if (ctx && ctx->layout) {
```

Comment:
Shouldn't we return an error if there's no layout (to keep the same semantics as previous code).

Reply:
inode_ctx_get will return 0 only if it has found layout, in case if no layout it will return non zero.

Reply:
No. At line 11208 `ret` will be reset to 0, but `ctx->layout` may be missing.

Reply:
i will update